server/services/wardrobe_gaps.py

```
@@ -0,0 +1,354 @@
+  - moderate:     only one item in a required category for the style, OR a
+                  recommended garment_type pairing exists in category_rules
+                  for the style but the user owns none of either side.
+  - nice_to_have: palette diversity gap derived from the style fingerprint's
+                  `dominant_palette` (e.g. no neutral tops for Formal). Skipped
+                  silently when the fingerprint is missing or low-confidence.
+"""
+
+from __future__ import annotations
+
+import logging
+from typing import Any, Iterable
```

Ruff (0.15.12)

[warning] Import from `collections.abc` instead: `Iterable`

Import from `collections.abc`

(UP035)